### PR TITLE
refactor: check the donator veOLAS balance as well

### PR DESCRIPTION
**Pull request by kupermind**
- Introduce a check for the donator `veOLAS` balance along with the service owner balance;
- Coverage is back to 100%.

### CODE REVIEW

File: contracts/GenericBondCalculator.sol [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.18;
 
-import {mulDiv} from "@prb/math/src/Core.sol";
+import {mulDiv} from "@prb/math/src/Common.sol";
```
Comment (kupermind) [inline contracts/GenericBondCalculator.sol:4]: This is due to the PRB Math library change.

File: contracts/Tokenomics.sol [hunk 2]
```diff
@@ -843,7 +846,7 @@ contract Tokenomics is TokenomicsConstants, IErrorsTokenomics {
         // Convert devsPerCapital
         UD60x18 fpDevsPerCapital = UD60x18.wrap(devsPerCapital);
         fp = fp.mul(fpDevsPerCapital);
-        UD60x18 fpNumNewOwners = toUD60x18(numNewOwners);
+        UD60x18 fpNumNewOwners = convert(numNewOwners);
```
Comment (kupermind) [inline contracts/Tokenomics.sol:849]: This is due to the PRB Math library change.

File: contracts/test/MockVE.sol [hunk 3]
```diff
@@ -3,14 +3,14 @@ pragma solidity ^0.8.18;
...
-    address[] public accounts;
     uint256 public balance = 50 ether;
     uint256 public supply = 100 ether;
     uint256 public weightedBalance = 10_000 ether;
+    mapping(address => uint256) public accountWeightedBalances;
 
     /// @dev Simulates a lock for the specified account.
     function createLock(address account) external {
-        accounts.push(account);
+        accountWeightedBalances[account] = weightedBalance;
```
Comment (kupermind) [inline contracts/test/MockVE.sol:13]: This is much better for testing the veOLAS mock.

**Review by mariapiamo — APPROVED** [review]

File: .solhint.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "extends": "solhint:recommended",
   "plugins": [],
   "rules": {
-    "compiler-version": ["warn", "^0.8.18"],
+    "compiler-version": ["warn", "^0.8.20"],
```
Comment (DavidMinarsch) [inline .solhint.json:5]: I read online that PUSH0 code causes incompatibilities of EVM on Ethereum and other L1s. We should probably read up on the issues. Various suggestions how to address this

File: contracts/TokenomicsConstants.sol [hunk 5]
```diff
@@ -8,7 +8,7 @@ import "@prb/math/src/UD60x18.sol";
 /// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
 abstract contract TokenomicsConstants {
     // Tokenomics version number
-    string public constant VERSION = "1.0.0";
+    string public constant VERSION = "1.0.1";
```
Comment (DavidMinarsch) [inline contracts/TokenomicsConstants.sol:11]: Are we redeploying this contract?

File: test/Dispenser.js [hunk 6]
```diff
@@ -258,6 +260,125 @@ describe("Dispenser", async () => {
             await snapshot.restore();
         });
 
+        it("Claim incentives for unit owners when donator and service owner are different accounts", async () => {
```
Comment (DavidMinarsch) [inline test/Dispenser.js:263]: Can we have some negative cases as well?

File: contracts/TokenomicsConstants.sol (hunk 5 above)
Comment (kupermind) [reply contracts/TokenomicsConstants.sol:11]: Yes, it's part of the Tokenomics contract.

File: .solhint.json (hunk 4 above)
Comment (77ph) [reply .solhint.json:5]: 100% correct. arbitrum as example not supported push0.
https://ethereum.stackexchange.com/questions/150281/invalid-opcode-opcode-0x5f-not-defined
https://developer.arbitrum.io/solidity-support
better not rush with shanghai as default EVM

Comment (kupermind) [reply .solhint.json:5]: As long as the chain runs the Shanghai upgrade, we are safe:
https://medium.com/coinmonks/push0-opcode-a-significant-update-in-the-latest-solidity-version-0-8-20-ea028668028a

https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html?ref=zaryabs.com#setting-the-evm-version-to-target

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**